csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/Assets.cs

```diff
@@ -143,14 +143,31 @@ public bool TryParse(string json, DependencyContainer dependencies)
             }
         }
 
+        private static bool TryReadAllText(string path, ProgressMonitor progressMonitor, out string content)
+        {
+            try
+            {
+                content = File.ReadAllText(path);
+                return true;
+            }
+            catch (Exception e)
+            {
+                progressMonitor.LogInfo($"Failed to read assets file '{path}': {e.Message}");
+                content = "";
```

Comment:
Would it make more sense to set this to `null` rather than the empty string? If someone ends up using this for something else in the future and forgets to inspect the boolean result to check for success, they could potentially get mislead into believing they successfully read an empty file.
```suggestion
                content = null;
```

Reply:
Oh wait, that won't work here. We don't want the out parameter to have type `string?`. Will revert. I suppose that if you use the output when the method returns false then you are on your own.

Reply:
Would adding `[NotNullWhen(returnValue: true)]` to the parameter solve this problem?

Reply:
TIL :-)

Reply:
There are a couple of these nullability analysis helping attributes. See the list here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis